src/main/java/ietf/params/xml/ns/icalendar/util/ICalendar.java

```diff
@@ -458,51 +458,21 @@
      * configuration.
      */
     switch (recurType.getFreq()) {
-      case SECONDLY:
-        dateSet.addAll(bySecond(asSet(periodStart), recurType, periodStart));
-        break;
-      case MINUTELY:
-        dateSet.addAll(bySecond(byMinute(asSet(periodStart), recurType, periodStart),
-                                recurType, periodStart));
-        break;
-      case HOURLY:
-        dateSet.addAll(bySecond(byMinute(byHour(asSet(periodStart), recurType, periodStart),
-                                         recurType, periodStart),
-                                recurType, periodStart));
-        break;
-      case DAILY:
-        dateSet.addAll(bySecond(byMinute(byHour(asSet(periodStart),
-                                                recurType, periodStart),
-                                         recurType, periodStart),
-                                recurType, periodStart));
-
-        break;
-      case WEEKLY:
-        dateSet.addAll(bySecond(byMinute(byHour(byDay(asSet(periodStart),
-                                                      recurType, periodStart),
-                                                recurType, periodStart),
-                                         recurType, periodStart),
-                                recurType, periodStart));
-
-        break;
-      case MONTHLY:
-        dateSet.addAll(bySecond(byMinute(byHour(byDay(byMonthDay(byMonth(asSet(periodStart), recurType, periodStart),
-                                                                          recurType, periodStart),
-//                                                               recurType, periodStart, weekFields),
-                                                      recurType, periodStart),
-                                                recurType, periodStart),
-                                         recurType, periodStart),
-                                recurType, periodStart));
-        break;
       case YEARLY:
-        dateSet.addAll(bySecond(byMinute(byHour(byDay(byWeekNo(byMonthDay(byMonth(byYearDay(asSet(periodStart), recurType, periodStart),
-                                                                                  recurType, periodStart),
-                                                                          recurType, periodStart),
-                                                               recurType, periodStart, weekFields),
-                                                      recurType, periodStart),
-                                                recurType, periodStart),
-                                         recurType, periodStart),
-                                recurType, periodStart));
+        dateSet = byYearDay(dateSet, recurType, periodStart);
+        dateSet = byWeekNo(dateSet, recurType, periodStart, weekFields);
+      case MONTHLY:
```

Comment:
The order in which the series expansions occur is important and must be preserved.